feat: add filtering to expenses

Edu58 opened the pull request.

Edu58 commented:
@mohitkhatri06  It is ready for review. You can test and give feedback

mohitkhatri06 commented:
after applying the filter add pagination, as for now it is taking long time to response after providing the range. And also if we can see the range what we had chosen (as for now after clicking the filter button we are not able to see which range I chosen), there should be some different button or something that can clear the filter. Let me know if you have any queries.

bharathkumarchowdary commented:
This filtering doesn't look good


![image](https://github.com/GVR-Secure-Solutions/lekkalu-frontend/assets/8872935/a4c2e80b-2569-4ee8-aaf1-9bc1dabebbf6)

bharathkumarchowdary commented:
> after applying the filter add pagination, as for now it is taking long time to response after providing the range. And also if we can see the range what we had chosen (as for now after clicking the filter button we are not able to see which range I chosen), there should be some different button or something that can clear the filter. Let me know if you have any queries.

I can't understand your comment @mohitkhatri06 , Can you please explain in layman terms?

mohitkhatri06 commented:
@bharathkumarchowdary as for now after we choose start and end date and click on the filter button, all records comes in a single page, whole list at once seems like.
Apart from that after selecting the range and click on the button it doesn't show that which range I have chosen or for which filter its showing the results.

ivangarcilazo reviewed (commented):
Yes, I think the same with Mohit and Bharath, just a improve in a design, I think like to make the form to row direction instead of column. And when I apply the filter, the dates are deleted, It's better if the dates selected dont are deleted, for better user experience. 👍

Edu58 commented:
@mohitkhatri06  Check the design now. For the pagination. I have to make changes to the API so it can serve paginated data also

mohitkhatri06 commented:
@Edu58 still when we enter the range and click on the filter button the selected range disappears.

Edu58 commented:
@mohitkhatri06 
@bharathkumarchowdary 
Fixed the clearing of the date range on filtering. Also added a clear button to clear the filters once done.

![fixedexpenses](https://github.com/GVR-Secure-Solutions/lekkalu-frontend/assets/79555732/eed92eb4-45c7-4da9-acbb-5053f6c729bd)

mohitkhatri06 commented on src/components/Expenses/Expenses.jsx:
```diff
@@ -134,38 +171,81 @@ const Expenses = ({ Context }) => {
+          >
+            <SkipNext />
+          </IconButton>
+          {page * 10 + 1} - {page * 10 + 10} of 70
```
instead of that 70 constant we should have the data which we get from the end point

mohitkhatri06 commented:
> @mohitkhatri06 Check the design now. For the pagination. I have to make changes to the API so it can serve paginated data also ![expensesfix](https://user-images.githubusercontent.com/79555732/252605907-a0a9e244-a69f-49c2-be44-c5e4e17b959a.png)

as we need to do some changes in pagination (backend) , approving the following PR,

ivangarcilazo reviewed (dismissed):
Everything looks good and functional.

ivangarcilazo reviewed (approved):
Works fine :)
